library/general/src/lib/ui/srv_status_component.rb

```diff
@@ -0,0 +1,236 @@
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact Novell about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "yast"
+Yast.import "Service"
+Yast.import "UI"
+
+module UI
+  # Component encapsulating the widgets for managing the status of services (both
+  # currently and on system boot) and the behavior associated to those widgets
+  #
+  # As long as #handle_input is invoked in the event loop, the component will
+  # handle interactive starting and stopping of the service on user demand. In
+  # addition #reload can be used after saving the settings.
+  #
+  # To manage the status on boot, the component can be queried for the user
+  # selection using #enabled?. In addition enabled_callback (in constructor)
+  # can be used to observe the status of the corresponding field in the UI.
+  class SrvStatusComponent
+    include Yast::UIShortcuts
+    include Yast::I18n
+    include Yast::Logger
+
+    # @param service_name [String] name of the service as expected by
+    #   Yast::Service
+    # @param reload [Boolean] initial value for the "reload" checkbox.
+    #   Keep in mind it will always be displayed as unchecked if the service is
+    #   not running, despite the real value.
+    # @param enabled_callback [Proc] callback executed when the "enabled on
+    #   boot" checkbox is changed. The only parameter of the callback is the new
+    #   state of the checkbox (boolean).
+    def initialize(service_name, reload: true, enabled_callback: nil)
+      @service_name = service_name
+      @reload = reload
+      @enabled_callback = enabled_callback
+
+      @enabled = service_enabled?
+      @id_prefix = "_srv_status_#{@service_name}"
+      textdomain "base"
+    end
+
+    # @return [YaST::Term]
+    def widget
+      Frame(
+        _("Service Status"),
+        VBox(
+          ReplacePoint(Id("#{id_prefix}_status"), status_widget),
+          reload_widget,
+          VSpacing(),
+          on_boot_widget
+        )
+      )
+    end
+
+    # Handles the input triggered by the widgets, this method must be called in
+    # the event loop of the dialog using the component.
+    def handle_input(input)
+      case input
+      when "#{id_prefix}_stop"
+        stop_service
+        refresh_widget
+      when "#{id_prefix}_start"
+        start_service
+        refresh_widget
+      when "#{id_prefix}_reload"
+        @reload = Yast::UI.QueryWidget(Id(input), :Value)
+      when "#{id_prefix}_enabled"
+        @enabled = Yast::UI.QueryWidget(Id(input), :Value)
+        @enabled_callback.call(@enabled) if @enabled_callback
```

Comment:
If there is Sandi Metz she will say you how she hates ifs :)...what is possible to do it to use EmptyCallback instead of nil in default constructor.